tests/server/listPuzzles.test.js

```diff
@@ -13,55 +14,81 @@ jest.mock("../../server/middleware", () => {
     }),
     logRouteAndCheckAuthorization: jest.fn((req, res, next) => {
       req.session.userID = 1;
-      req.session.username = 'alice';
+      req.session.username = "alice";
       next();
     })
   }
-})
-
-/**
- * Helper function to insert test puzzles into db
- * @param {int} k - Number of test puzzles to insert
- * @returns nothing
- */
-function dataHelper(k) {
-  db.run(
-    `INSERT INTO Puzzle VALUES (1, "test puzzle 1", "test description 1")`
-  );
-
-  db.run(`INSERT INTO User_Puzzle VALUES (1, 1, NULL, 1)`)
-}
-
-// block console logging, generate mock data
-beforeAll(() => {
-  jest.spyOn(console, "log").mockImplementation(() => { });
-  jest.spyOn(console, "error").mockImplementation(() => { });
-  dataHelper();
 });
 
-describe("Puzzles endpoint", () => {
-  test("Response 200 - successful query", async () => {
-    const res = await request(app).get("/api/listPuzzles");
-    console.log(res.body)
+describe("Tests for GET at /api/listPuzzles", () => {
+  const route = "/api/listPuzzles";
+  const rows = [
+    {
+      puzzle_id: 1,
+      title: "Title",
+      description: "Description"
+    },
+    {
+      puzzle_id: 2,
+      title: "Title 2",
+      description: "Description 2"
+    }
+  ]
 
-    expect(res.statusCode).toEqual(200); // status code should be 200
-    expect(res.body).toEqual(
+  const userPuzzleCompletion = [
+    {
+      progress: 1,
+      puzzle_id: 1
+    }
+  ]
+
+  beforeAll(() => {
+    jest.spyOn(console, "log").mockImplementation();
+    jest.spyOn(console, "error").mockImplementation();
+  });
+
+  test("200 - Success", async () => {
+    jest.spyOn(db, "all").mockImplementationOnce((query, callback) => {
+      callback(null, rows)
+    })
+
+    jest.spyOn(db, "all").mockImplementationOnce((query, callback) => {
```

Comment:
Any reason for `jest.spyOn` vs. `db.all = jest.fn(...)` as was used before?

Reply:
(seen at lines 72, 82 and 86)

Reply:
Ok now you have one of them as `db.all = jest.fn()` but then the other as `jest.spyOn`????

Reply:
Nvm you only changed one of them.,.